### PR TITLE
new docs: split files

### PR DESCRIPTION
**Pull request by matallo**
I've splitted the `API.md` in several files in order to have them better work with new docs

does this structure fits well?

also, I've spotted some legacy (?) build tasks and files:

- `Makefile`
- `doc_template.html`
- `style.css`
- `thumbnail.png`

shall we get rid of these?

CR @alonsogarciapablo

### CODE REVIEW

File: doc/other_stuff.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+sql_api_template: 'https://{user}.test.com'
+{% endhighlight %}
+
+CartoDB.js will replace ``{user}``.
+
+Notice that you don't need to set the path to the endpoint, CartoDB.js will set it automatically.
+
+## Bounds wrapper
+
+We have added an easy method to get the bounding box for any dataset or filtered query using the CartoDB.js library. The **getBounds** function can be useful for guiding users to the right location on a map or for loading only the right data at the right time based on user actions.
```
Comment (alonsogarciapablo) [inline doc/other_stuff.md]: This is also documented [here](https://github.com/CartoDB/cartodb.js/pull/725/files#diff-363d4745c1355c9340fcee8e9791e493R56). Perhaps we can remove this and complete the description (if needed on the other file)

File: doc/other_stuff.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
+# Other important stuff
```
Comment (alonsogarciapablo) [inline doc/other_stuff.md]: Not sure if reviewing the docs should also be part of this PR but I think we could try to get rid of this whole file and move its content to the other sections.

Comment (matallo) [inline doc/other_stuff.md]: We definitely need to review the docs, at least to check the styles are rendering properly

/cc @csobier

**Comment by alonsogarciapablo** [comment]
Structure makes sense since it matches the existing section. I left a comment about trying to get rid of the `other_stuff.md` file. About the "legacy" files, they are pretty old, so I would :fire:. Good catch!

:+1:

File: doc/other_stuff.md (hunk 1 above)
Comment (matallo) [inline doc/other_stuff.md]: removed here https://github.com/CartoDB/cartodb.js/commit/581dd968cfe4c5e24d23c23321d16326c17b7597

@csobier extend the description if needed

**Comment by matallo** [comment]
@csobier what about `raster_howto.md`, shall we find some place for it in the docs or ditch it completely?

I'm leaving it for now

**Comment by matallo** [comment]
the same with `metrics.md`, this isn't covered in the public docs, what should we do here @csobier

**Comment by matallo** [comment]
created specific issues for those
https://github.com/CartoDB/docs/issues/449
https://github.com/CartoDB/docs/issues/448

**Comment by viddo** [comment]
re: the Makefile I've planned to review it in [#722 cleanup post-add-browserify](https://github.com/CartoDB/cartodb.js/issues/722), but if it can just be removed earlier on better (as long as it doesn't imply changes to [gruntfile.js](https://github.com/CartoDB/cartodb.js/blob/v4-browserify/Gruntfile.js) which I already modified quite a bit).

**Comment by matallo** [comment]
nah, it's just removing non-used build scripts and files, I won't touch the [gruntfile.js](https://github.com/CartoDB/cartodb.js/blob/v4-browserify/Gruntfile.js)

**Comment by matallo** [comment]
also, I spotted a `quickstart.md` that could easily be removed, can you confirm @alonsogarciapablo @viddo @csobier

**Comment by alonsogarciapablo** [comment]
:fire:

**Comment by matallo** [comment]
:sparkles:

**Pull request merged**